nomad/core_sched.go

```diff
@@ -152,6 +155,11 @@ OUTER:
 	c.logger.Debug("job GC found eligible objects",
 		"jobs", len(gcJob), "evals", len(gcEval), "allocs", len(gcAlloc))
 
+	// Clean up any outstanding volume claims
+	if err := c.volumeReap(gcJob, eval.LeaderACL); err != nil {
```

Comment:
I originally thought we wanted to do this in the `jobReap` as part of the `BatchDeregister` it calls, but the evals for reaping allocs get emitted first and that'd cause a race where we're likely to end up with dangling volume claims.